src/test/java/io/appium/java_client/android/AndroidDriverTest.java

```diff
@@ -140,4 +141,33 @@
         assertNotNull(driver.getDisplayDensity());
         assertNotEquals(0, driver.getSystemBars().size());
     }
+
+    @Test public void getSupportedPerformanceDataTypesTest() {
+        driver.startActivity("io.appium.android.apis", ".ApiDemos");
+
+        List<String> supportedPerformanceDataTypes = driver.getSupportedPerformanceDataTypes();
+        assert(supportedPerformanceDataTypes.size() == 4);
+
+    }
+
+    @Test public void getPerformanceDataTest() {
+        driver.startActivity("io.appium.android.apis", ".ApiDemos");
+
+        List<String> supportedPerformanceDataTypes = driver.getSupportedPerformanceDataTypes();
+
+        for(int i = 0 ; i < supportedPerformanceDataTypes.size() ;  ++ i){
+
+            String dataType = supportedPerformanceDataTypes.get(i);
+
+            List<List<Object>> valueTable = driver.getPerformanceData("com.example.android.apis", dataType, 60000);
+
+            int valueTableHeadLength = valueTable.get(0).size();
+
+            for(int j = 1 ; j < valueTable.size() ; ++ j){
```

Comment:
I am not sure about the difference between List.get(i) and List.subList(i,i).
but I applied it.

and one more, 
in case of this method, 
there is no reason to check the head value of the returned table, because the implementation is dependent for the platform that means the head value could be changed.